src/wazuh_modules/syscollector/tests/sysCollectorFlatbuffers/sysCollectorFlatbuffers_test.cpp

```diff
@@ -0,0 +1,424 @@
+    "    os_platform:string;"
+    "    sysname:string;"
+    "    release:string;"
+    "    version:string;"
+    "    os_release:string;"
+    "    os_display_version:string;"
+    "    triaged:int;"
+    "    reference:string;"
+    "    checksum:string;"
+    "}"
+    "root_type Sys_osinfo;"};
+
+const std::string hwInfoSchemaStr
+{
+    "namespace Syscollector;"
+    "table Sys_hwinfo {"
+    "    scan_time:string;"
+    "    board_serial:string;"
+    "    cpu_name:string;"
+    "    cpu_cores:int;"
+    "    cpu_MHz:double;"
+    "    ram_total:int;"
+    "    ram_free:int;"
+    "    ram_usage:int;"
+    "    checksum:string;"
+    "}"
+    "root_type Sys_hwinfo;"};
+
+const std::string portsSchemaStr
+{
+    "namespace Syscollector;"
+    "table Sys_ports {"
+    "    scan_time:string;"
+    "    protocol:string;"
+    "    local_ip:string;"
+    "    local_port:int;"
+    "    remote_ip:string;"
+    "    remote_port:int;"
+    "    tx_queue:int;"
+    "    rx_queue:int;"
+    "    inode:long;"
+    "    state:string;"
+    "    PID:int;"
+    "    process:string;"
+    "    item_id:string;"
+    "    checksum:string;"
+    "}"
+    "root_type Sys_ports;"};
+
+const std::string programsSchemaStr
+{
+    "namespace Syscollector;"
+    "table Sys_programs {"
+    "    scan_time:string;"
+    "    format:string;"
+    "    name:string;"
+    "    priority:string;"
+    "    section:string;"
+    "    size:int;"
+    "    vendor:string;"
+    "    install_time:string;"
+    "    version:string;"
+    "    architecture:string;"
+    "    multiarch:string;"
+    "    source:string;"
+    "    description:string;"
+    "    location:string;"
+    "    cpe:string;"
+    "    msu_name:string;"
+    "    checksum:string;"
+    "    item_id:string;"
+    "}"
+    "root_type Sys_programs;"};
+
+const std::string hotfixesSchemaStr
+{
+    "namespace Syscollector;"
+    "table Sys_hotfixes {"
+    "    scan_time:string;"
+    "    hotfix:string;"
+    "    checksum:string;"
+    "}"
+    "root_type Sys_hotfixes;"};
+
+const std::string processesSchemaStr
+{
+    "namespace Syscollector;"
+    "table Sys_processes {"
+    "    scan_time:string;"
+    "    pid:string;"
+    "    name:string;"
+    "    state:string;"
+    "    ppid:int;"
+    "    utime:int;"
+    "    stime:int;"
+    "    cmd:string;"
+    "    argvs:string;"
+    "    euser:string;"
+    "    ruser:string;"
+    "    suser:string;"
+    "    egroup:string;"
+    "    rgroup:string;"
+    "    sgroup:string;"
+    "    fgroup:string;"
+    "    priority:int;"
+    "    nice:int;"
+    "    size:int;"
+    "    vm_size:int;"
+    "    resident:int;"
+    "    share:int;"
+    "    start_time:long;"
+    "    pgrp:int;"
+    "    session:int;"
+    "    nlwp:int;"
+    "    tgid:int;"
+    "    tty:int;"
+    "    processor:int;"
+    "    checksum:string;"
+    "}"
+    "root_type Sys_processes;"};
+
+bool parseJSON(const std::string& schemaPath, const std::string& jsonStr)
```

Comment:
This method checks that the incoming JSON could be parsed according to the flatbuffer schema provided to the parser.

In my opinion, the second part of this method (the "GenText" call), doesn't make much sense if we don't compare the JSON received by the method with the one generated by the flatbuffer function "GenText".

Here is an internet example: https://github.com/google/flatbuffers/blob/master/samples/sample_text.cpp

Reply:
Totally unnecessary, removed.